src/web/gwc/src/main/java/org/geoserver/gwc/web/layer/UnconfiguredCachedLayersProvider.java

```diff
@@ -62,6 +75,80 @@ public int compare(TileLayer o1, TileLayer o2) {
     static final List<Property<TileLayer>> PROPERTIES =
             Collections.unmodifiableList(Arrays.asList(TYPE, NAME, ENABLED));
 
+    private static final String KEY_SIZE = "key.size";
+    private final Cache<String, Integer> cache;
+
+    private GWCConfig defaults;
+
+    public UnconfiguredCachedLayersProvider() {
+        super();
+        // Initialization of an inner cache in order to avoid to calculate several times
+        // the size() method in a time minor than a second
+        CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
+        cache = builder.expireAfterWrite(1, TimeUnit.SECONDS).build();
+
+        defaults = GWC.get().getConfig().saneConfig().clone();
+        defaults.setCacheLayersByDefault(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Provides a page of transient TileLayers for the LayerInfo and LayerGroupInfo objects in
+     * Catalog that don't already have a configured TileLayer on their metadata map and that match
+     * the page {@link #getFilter() filter}, in the specified {@link #getSort() sort order}.
+     */
+    @Override
+    public Iterator<TileLayer> iterator(long first, long count) {
+        final SortParam<Object> sort = getSort();
+        final Filter filter = getFilter();
+        final Stream<TileLayer> stream;
+        if (sort == null) {
+            stream =
+                    unconfiguredLayers(filter)
+                            .skip(first)
+                            .limit(count)
+                            .map(this::createUnconfiguredTileLayer);
+        } else {
+            Comparator<TileLayer> comparator = getComparator(sort);
+            stream =
+                    unconfiguredLayers(filter)
+                            .map(this::createUnconfiguredTileLayer)
+                            .sorted(comparator)
+                            .skip(first)
+                            .limit(count);
+        }
+        return new CloseableIteratorAdapter<TileLayer>(stream.iterator(), () -> stream.close());
```

Comment:
Hum... works for the time being but have a couple of hesitations (just thiking out loud here):
1) Wicket is in charge of using the iterator here, will it recognize the "closeable" and call to close it (not sure, maybe not)?
2) The adapter closes when done (good), uses the finalizer as a fallback to eventually close it otherwise (exception for example), but that is not reliable and it's deprecated since Java 9.
Eventually we'll need a better way.